src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c

```diff
@@ -21,47 +48,601 @@ const ActorInit Bg_Po_Event_InitVars = {
+static s16 blocksX[] = { 0x0865, 0x07B1, 0x0775 };
+static s16 blocksZ[] = { 0xFA7E, 0xFABA, 0xFA06 };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+static s32 firstFall = 0;
+static f32 blockPushDist = 0.0f;
+static u8 puzzleState;
+
+void BgPoEvent_InitPaintings(BgPoEvent* this, GlobalContext* globalCtx) {
+    ColliderTrisItemInit* item;
+    Vec3f* vtxVec;
+    s32 i1;
+    s32 i2;
+    Vec3f sp9C[3];
+    f32 coss;
+    f32 sins;
+    f32 scaley;
+    s32 phi_t2;
+
+    sins = Math_Sins(this->dyna.actor.shape.rot.y);
+    coss = Math_Coss(this->dyna.actor.shape.rot.y);
+    if (this->actorType == 4) {
+        sins *= 2.4f;
+        scaley = 1.818f;
+        coss *= 2.4f;
+    } else {
+        scaley = 1.0f;
+    }
+    for (i1 = 0; i1 < sTrisInit.count; i1++) {
+        item = &sTrisInit.list[i1];
+        if (1) {} // This section looks like a macro of some sort.
+        for (i2 = 0; i2 != 3; i2++) {
+            vtxVec = &item->dim.vtx[i2];
+            sp9C[i2].x = (vtxVec->x * coss) + (this->dyna.actor.initPosRot.pos.x + (sins * vtxVec->z));
+            sp9C[i2].y = (vtxVec->y * scaley) + this->dyna.actor.initPosRot.pos.y;
+            sp9C[i2].z = this->dyna.actor.initPosRot.pos.z + (coss * vtxVec->z) - (vtxVec->x * sins);
+        }
+        func_800627A0(&this->collider1, i1, &sp9C[0], &sp9C[1], &sp9C[2]);
+    }
+    if (this->actorType != 4) {
+        if (this->actorIndex != 2) {
+            if (this->actorType == 2) {
+                phi_t2 = this->actorIndex;
+            } else {
+                phi_t2 = this->actorIndex + 2;
+            }
+            if (Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_BG_PO_EVENT,
+                                   paintingsX[phi_t2], paintingsY[this->actorIndex], paintingsZ[this->actorIndex], 0,
+                                   this->dyna.actor.shape.rot.y + 0x8000, 0,
+                                   ((this->actorIndex + 1) << 0xC) + (this->actorType << 8) +
+                                       this->dyna.actor.params) == 0) {
+                Actor_Kill(&this->dyna.actor);
+                return;
+            }
+            if (this->actorIndex == 0) {
+                if (this->dyna.actor.child->child == NULL) {
+                    Actor_Kill(&this->dyna.actor);
+                    return;
+                }
+                this->dyna.actor.parent = this->dyna.actor.child->child;
+                this->dyna.actor.child->child->child = &this->dyna.actor;
+            }
+        }
+    }
+    this->timer = 0;
+    if (this->actorType == 4) {
+        puzzleState = 0;
+        this->actionFunc = &BgPoEvent_AmyWait;
+    } else {
+        puzzleState = (s32)(Math_Rand_ZeroOne() * 3.0f) % 3;
+        this->actionFunc = &BgPoEvent_PaintingEmpty;
+    }
+}
+
+void BgPoEvent_InitBlocks(BgPoEvent* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
```

Comment:
Do you really have to use thisx over this?

Reply:
No, but it's a lot shorter than having this->dyna.actor everywhere and the stack padding is needed.